### PR TITLE
TEET-399 project user mgmt

### PR DESCRIPTION
**Pull request by tatut**
Ability to set permission role when giving out project permissions.
Also ability to create new empty user if it doesn't already exist.

### CODE REVIEW

File: app/frontend/src/cljs/teet/project/project_controller.cljs [hunk 1]
```diff
@@ -614,12 +614,17 @@
 
   SaveProjectPermission
   (process-event [{project-id :project-id form-data :form-data} app]
-    (let [participant (:project/participant form-data)]
+    (let [participant (:project/participant form-data)
+          role (:permission/role form-data)]
+      (log/info "SAVE PROJECT PErmisSION " form-data)
```
Comment (outamaa) [inline app/frontend/src/cljs/teet/project/project_controller.cljs]: Nitpick: Why PErmisSION?

File: app/common/src/cljc/teet/authorization/authorization_check.cljc [hunk 2]
```diff
@@ -14,6 +15,12 @@
                                  slurp
                                  read-string))))
 
+(defonce roles
```
Comment (outamaa) [inline app/common/src/cljc/teet/authorization/authorization_check.cljc]: Maybe rename `all-roles`, doc string "A delay of a set of all role keywords used in the application." or some such.

File: app/frontend/src/cljs/teet/project/project_view.cljs [hunk 3]
```diff
@@ -221,14 +221,33 @@
...
-  [:div
-   [form/form {:e! e!
-               :value user
-               :on-change-event #(e! (project-controller/->UpdateProjectPermissionForm %))
-               :save-event #(e! (project-controller/->SaveProjectPermission project-id user))
-               :spec :project/add-permission-form}
-    ^{:attribute :project/participant}
-    [select/select-user {:e! e! :attribute :project/participant}]]])
+  (let [roles (-> @authorization-check/roles
+                  (disj :admin)
```
Comment (outamaa) [inline app/frontend/src/cljs/teet/project/project_view.cljs]: Is this enforced in the backend as well?

Comment (tatut) [reply app/frontend/src/cljs/teet/project/project_view.cljs]: I meant to put this pr down as WIP, the authorization matrix needs permissions on who can give out what permission.

Comment (outamaa) [reply app/frontend/src/cljs/teet/project/project_view.cljs]: Roger that.

Comment (tatut) [reply app/frontend/src/cljs/teet/project/project_view.cljs]: For now, added assertion on backend side that admin can't be given.

**Pull request merged**